### PR TITLE
Add support for theme on macOS, and unify theme methods

### PR DESCRIPTION
**Pull request by keiya01**
- [x] Tested on all platforms changed
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [x] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [x] Created or updated an example program if it would help users understand this functionality
- [x] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

## How to test

- `WindowBuilderExtMacOS::with_theme` should force theme even if theme is changed by system preference
- `WindowExtMacOS::theme` should return current theme even if theme is changed.
- `ThemeChanged` event should be happen whenever theme is changed.

### CODE REVIEW

File: src/platform_impl/macos/window_delegate.rs
Comment (madsmtm) [inline src/platform_impl/macos/window_delegate.rs:617]: Remove debug logging.

**Review by madsmtm — CHANGES_REQUESTED** [review]
Thanks, implementation looks good!

Given that we now have three platforms (macOS, Windows and Wayland) that support querying and setting themes, perhaps it would make sense to unify these, and add an example for it; but that can be done later.

**Review by madsmtm — APPROVED** [review]
Cool, thanks! (Will merge once CI finishes)

File: src/platform/macos.rs
Comment (madsmtm) [reply src/platform/macos.rs:129]: I note this above as well, but decided to just postpone it until someone put in the work to make a PR for it.

Comment (kchibisov) [reply src/platform/macos.rs:129]: I mean that you don't need that much skills for that, basically grep all using `Theme` and do move.

File: src/platform/macos.rs
Comment (keiya01) [reply src/platform/macos.rs:41]: > Could we not do that as a macOS extension? We have theme setting on like every platform

If I do this, I need to implement theme method for getting current theme on unix?
I think this feature is not implemented on unix platform.

> And should likely return Option in that case, since we could fail identify it? Like on old macOS.

This should be handled by https://github.com/rust-windowing/winit/pull/2441/files#diff-4803bb2335a17d700335730b9f62828e001faa602b9b894dff572422c737b78aR1364-R1367.

File: src/platform/macos.rs
Comment (keiya01) [reply src/platform/macos.rs:129]: How do we handle wayland and x11 platform? These have different interface about theme.

Comment (kchibisov) [reply src/platform/macos.rs:129]: X11 leave unimplemented and reference the gtk variant. Wayland, macOS, and windows should use the same `with_theme`. For the getter you should return the `None` for Wayland and windows if you don't plan to add it.

You don't have to implement anything here, just add `with_theme` on `WindowBuilder` and `theme` on `Window` that uses existing impls from the extensions.

File: src/window.rs
Comment (madsmtm) [inline src/window.rs:1313]: I know that while most users use a light theme, I don't actually think a default makes sense here?

File: src/platform/x11.rs
Comment (madsmtm) [inline src/platform/x11.rs:263]: Nit: Remove

File: CHANGELOG.md
Comment (madsmtm) [inline CHANGELOG.md:13]: Don't worry about the changelog, I'll take care of it before merge

**Review by madsmtm — CHANGES_REQUESTED** [review]
Thanks for that, that is indeed nicer!

File: src/window.rs
Comment (keiya01) [reply src/window.rs]: I fixed this here.
[fix: remove unnecessary default for theme](https://github.com/rust-windowing/winit/pull/2441/commits/42c7318ca004d8bdda7a73cb3a5cf31fb92836fb)

File: src/platform/x11.rs
Comment (keiya01) [reply src/platform/x11.rs]: [refactor: remove unnecessary cfg](https://github.com/rust-windowing/winit/pull/2441/commits/f565a87b7c89505f4f907be3ec4bf314b641f109)
I fixed this here

**Review by msiglreith — APPROVED** [review]
A note in the changelog should be added that the `WindowExtWindows::theme` got replaced by `Window::theme` and `WindowBuilderExtWindows::with_theme` by `WindowBuilder::with_theme`

**Review by madsmtm — COMMENTED** [review]
@kchibisov does it make sense to merge `WindowBuilderExtWayland::with_wayland_csd_theme` with `WindowBuilder::with_theme`?

**Comment by kchibisov** [comment]
> @kchibisov does it make sense to merge WindowBuilderExtWayland::with_wayland_csd_theme with WindowBuilder::with_theme?

Yes. And we can add a comment that it controls only CSD. That's perfectly fine.

**Comment by keiya01** [comment]
I added comment about CSD!

**Review by madsmtm — APPROVED** [review]
I updated the macOS impl since the changes in #2465, and while I was at it, removed the remaining `WindowBuilderExtWayland::with_wayland_csd_theme` and updated the changelog as per requested by @msiglreith

File: src/window.rs
Comment (kchibisov) [inline src/window.rs:947]:
```suggestion
    /// Returns the current window theme.
    ///
    /// ## Platform-specific
```

**Review by kchibisov — CHANGES_REQUESTED** [review]
Besides that small doc change it should be good to go, though it requires a rebase.

**Pull request merged**
